valhalla/mjolnir/node_expander.h

```diff
@@ -65,6 +65,14 @@ struct Edge {
   // index of the target (end) node of the edge
   uint32_t targetnode_;
 
+  /**
+   * For now you cant be valid if you dont have any shape
+   * @return true if the edge has at least 1 shape point
+   */
+  bool is_valid() const {
+    return attributes.llcount > 0;
```

Comment:
in the loop where we create edges in the graph, we only add the edge in the next iteration of the loop. because of this there is always a previous edge that we are tracking but we only want to add it if its not the first previous edge (ie not a real edge). this little function helps make that easier to see. when you initialize an edge without actual data you end up with an edge that has no shape (nodes associated) yet so this function uses that to decide whether the edge is valid or not